contracts/FT-voting-simple/ProposalTarget.sol

```diff
@@ -16,16 +23,23 @@ contract ProposalTarget {
         _;
     }
 
-    constructor() {
-        governor = RIFGovernorFT(payable(msg.sender));
+    constructor(IGovernor _governor) {
+        _setGovernor(_governor);
     }
 
-    function onProposalExecution(uint256 _proposalId) public governorOnly {
-        // for some reason these commented lines cause tx rejection on RSK, but not on others
-        /* require(
+    function setGovernor(IGovernor _governor) external onlyOwner {
```

Comment:
Where is the owner address used by `onlyOwner` being set? (I do not see that in the constructor)

Also, consider that in the context of being used in a DAO, there should not be a single address (either EOA or SCA) that has "admin" privileges. Instead, the DAO itself should fulfil that responsibility ... so in terms of code, that would mean:
- removing `@openzeppelin/contracts/access/Ownable.sol` altogether
- replacing any usage of `onlyOwner` with `onlyGovernor`
- the constructor can remain as is, but `setGovernor` should only be callable by its current governor, with a check on `governor.state(_proposalId)`, just like any other proposal